File: CS/NorthwindObjects.cs
namespace Northwind {
    using System;
    using DevExpress.Xpo;
    
    
	[Persistent("Customers")]
	public class Customer : XPLiteObject {
		[Key]
		public string CustomerID;
		public string CompanyName;
		public string ContactTitle;

		[Association("CustomerOrders", typeof(Order))]
		public XPCollection Orders {
			get {
				return GetCollection("Orders");
			}
		}
	}

	[Persistent("Orders")]
	public class Order : XPLiteObject {
		[Key]
		public int OrderID;
		public DateTime ShippedDate;
		
		[Persistent("CustomerID"), Association("CustomerOrders")]
		public Customer Customer;

		[Persistent("EmployeeID"), Association("EmployeeOrders")]
		public Employee Employee;

        public string ShipName;
 	}

	[Persistent("Employees")]
	public class Employee : XPLiteObject {
		[Key]
		public int EmployeeID;
		public string FirstName;
		public string LastName;

		[Association("EmployeeOrders", typeof(Order))]
		public XPCollection Orders {
			get {
				return GetCollection("Orders");
			}
		}
	}
}

File: VB/Distinct/NorthwindObjects.vb
Imports System
Imports DevExpress.Xpo

Namespace Northwind

    <Persistent("Customers")>
    Public Class Customer
        Inherits XPLiteObject

        <Key>
        Public CustomerID As String

        Public CompanyName As String

        Public ContactTitle As String

        <Association("CustomerOrders", GetType(Order))>
        Public ReadOnly Property Orders As XPCollection
            Get
                Return GetCollection("Orders")
            End Get
        End Property
    End Class

    <Persistent("Orders")>
    Public Class Order
        Inherits XPLiteObject

        <Key>
        Public OrderID As Integer

        Public ShippedDate As Date

        <Persistent("CustomerID"), Association("CustomerOrders")>
        Public Customer As Customer

        <Persistent("EmployeeID"), Association("EmployeeOrders")>
        Public Employee As Employee

        Public ShipName As String
    End Class

    <Persistent("Employees")>
    Public Class Employee
        Inherits XPLiteObject

        <Key>
        Public EmployeeID As Integer

        Public FirstName As String

        Public LastName As String

        <Association("EmployeeOrders", GetType(Order))>
        Public ReadOnly Property Orders As XPCollection
            Get
                Return GetCollection("Orders")
            End Get
        End Property

        <PersistentAlias("[<Customer>][[<Order>][Customer = ^.This && Employee = ^.^.This]].Count()")>
        Public ReadOnly Property DistinctCustomerCount As Integer
            Get
                Return Convert.ToInt32(EvaluateAlias("DistinctCustomerCount"))
            End Get
        End Property
    End Class
End Namespace
